lib/oneview-sdk/image-streamer/client.rb

```diff
@@ -0,0 +1,154 @@
+# (C) Copyright 2016 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+require 'logger'
+require_relative '../config_loader'
+require_relative '../rest'
+require_relative '../ssl_helper'
+
+module OneviewSDK
+  module ImageStreamer
+    # The client defines the connection to the Image Streamer server and handles communication with it.
+    class Client
+      attr_reader :url, :token, :max_api_version
+      attr_accessor :ssl_enabled, :api_version, :logger, :log_level, :cert_store, :print_wait_dots, :timeout
```

Comment:
Sweet! You'd already done the work copying things over so it won't save us work right now, but later on down the road it'll be nice to only have to manage that shared code in 1 place.